src/api/users/src/routes/users.js

```diff
@@ -28,27 +28,36 @@ router.get('/:id', validateId(), async (req, res, next) => {
 });
 
 // get all users
-// rejected if the user collection is empty, otherwise users returned
-router.get('/', async (req, res, next) => {
-  try {
-    const usersRef = db.collection('users');
-    const snapshot = await usersRef.get();
+// rejected if the users collection is empty
+// otherwise n (specified via params) users are returned
+router.get('/', validatePagingParams(), async (req, res, next) => {
+  // if req.query.per_page is < 1 or unparsable, set to 100, otherwise set to the min of either req.query.page or 100
+  const parsePerPage =
+    req.query.per_page < 1 ? 100 : Math.min(parseInt(req.query.per_page, 10), 100);
+  const perPage = !parsePerPage || parsePerPage < 1 ? 100 : parsePerPage;
+  // if req.query.page is 0 or unparsable set to 1, otherwise set to the req.query.page
```

Comment:
It can't be 0 now, based on your validation rules, this comment should be updated.